quic/s2n-quic-core/src/buffer/receive_buffer.rs

```diff
@@ -269,7 +286,67 @@ impl ReceiveBuffer {
 
         self.allocate_request(0, request);
 
-        self.check_consistency();
+        self.invariants();
+
+        Ok(())
+    }
+
+    /// Advances the read and write cursors and discards any held data
+    ///
+    /// This can be used for copy-avoidance applications where a packet is received in order and
+    /// doesn't need to be stored temporarily for future packets to unblock the stream.
+    #[inline]
+    pub fn skip(&mut self, len: usize) -> Result<(), ReceiveBufferError> {
+        // zero-length skip is a no-op
+        if len == 0 {
+            return Ok(());
+        }
```

Comment:
Is this an appropriate use of `ensure!`, or should it just be for error cases?
```suggestion
        ensure!(len > 0, Ok(()));
```

Reply:
Funny you mention it.. I went back and forth on using it for this. Didn't know if it would be confusing to return `Ok` or not. What are your thoughts?